dspace-uclouvain/src/main/java/org/dspace/content/authority/UCLAuthority.java

```diff
@@ -0,0 +1,131 @@
+        DialPerson[] personsList = getStudentList(query);
+        Choice choices_list[] = new Choice[personsList.length];
+        for(int i = 0;i < personsList.length; i++){
+            DialPerson currentPerson = personsList[i];
+            Map<String, String> extras = generateExtras(currentPerson);
+            choices_list[i] = new Choice("UCLAuthority", currentPerson.getFullName(), currentPerson.getFullName(), extras);
+        }
+        return new Choices(choices_list, 0, choices_list.length, Choices.CF_AMBIGUOUS, false, -1);
+    }
+
+    // Not implemented 
+    @Override
+    public Choices getBestMatch(String text, String locale) {
+        return new Choices(Choices.CF_NOTFOUND);
+    }
+
+    @Override
+    public String getLabel(String key, String locale) {
+        return "";
+    }
+
+    @Override
+    public String getPluginInstanceName() {
+        return pluginInstanceName;
+    }
+
+    @Override
+    public void setPluginInstanceName(String name) {
+        this.pluginInstanceName = name;
+    }
+
+    // Retrieve the list of student based on the query (first and last name)
+    public DialPerson[] getStudentList(String query) {
+        // Requesting the list of similar student based on "query", taking only "students"
+        return this.uclClient.getStudentByTermWithFilter(query, "student");
+    }
+
+    // Generate a list of extra information that will be displayed under the main information 
+    // Also used to populate the fields when the user selects a choice in the list
+    private Map<String, String> generateExtras(DialPerson person){
+        Map<String, String> extras = new HashMap<>();
+
+        String entity = person.getEntity();
+        String orcid_id = person.orcid_id;
+        String employeeId = person.getEmployeeId();
+        String institution = person.getInstitution();
+
+        extras.put("institution-affiliation-name", institution);
+
+        // Managing data to be encoded in form fields
+        extras.put("data-authors_institution_code", institution);
+        extras.put("data-authors_email", person.getEmail());
+
+        if(!entity.isEmpty()){
+            extras.put("institution-entity-name", entity);
+        }
+        if(!orcid_id.isEmpty()){
+            extras.put("person_identifier_orcid", extractOrcidId(orcid_id, "https://orcid.org/"));
+        }
+
+        // FORM DATA COMPLETER => POPULATE FIELDS
+        if(institution.contains("UCL")){
+            extras.put("data-authors_identifier_fgs", person.getPrimaryId());
+        }
+        // Check if noma (employee_id) is present
+        if(!employeeId.isEmpty()){
+            extras.put("data-authors_identifier_noma", employeeId);
+        }
+
+        return extras;
+    }
+
+    // Retrieve the orcid id from an orcid string (ex: https://orcid.org/0000-0001-8304-5658 => 0000-0001-8304-5658)
+    private String extractOrcidId(String orcidLink, String pattern){
+        return orcidLink.replace(pattern, "");
+    }
```

Comment:
Seems a little bit overkill ?

Reply:
or create an `org.dspace.model.Orcid` class containing "getURI()" and "getID()" methods that could be use in future devs